pupil_src/shared_modules/surface_tracker/gui.py

```diff
@@ -380,9 +380,12 @@ def _draw_heatmap(self, surface):
 
     def on_click(self, pos, button, action):
         pos = np.array(pos, dtype=np.float32)
-        self._on_click_menu_buttons(action, pos)
-        self._on_click_corner_handles(action, pos)
-        self._on_click_marker_toggles(action, pos)
+        click_handlers_sorted_by_precedence = [
+            self._on_click_corner_handles,
+            self._on_click_menu_buttons,
+            self._on_click_marker_toggles,
+        ]
+        _ = any(handler(action, pos) for handler in click_handlers_sorted_by_precedence)
```

Comment:
I think using the early-exit implementation of `any()` is misleading here. This is much clearer in its intent I feel:
```python
for handler in click_handlers_sorted_by_precedence:
    if handler(action, pos):
        return
```

Reply:
I agree with the point about not being very clear in the intent, but I was following precedent set in `world.py` ([1][1], [2][2], [3][3], [4][4]), `player.py` ([5][5], [6][6], [7][7], [8][8]) and `eye.py` ([9][9]). I think we should settle on one way to describe this concept (early exist on event consumption) and use it consistently, whichever it is.

@papr thoughts?

[1]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/world.py#L353
[2]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/world.py#L671
[3]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/world.py#L676
[4]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/world.py#L681
[5]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/player.py#L212
[6]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/player.py#L630
[7]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/player.py#L635
[8]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/player.py#L640
[9]: https://github.com/pupil-labs/pupil/blob/master/pupil_src/launchables/eye.py#L244

Reply:
The fact that *ALL* those references have a two-line comment preceding, which explains what is happening afterwards with the `any()` confirms my suspicion 😃 

> Explicit is better than implicit.
-- https://www.python.org/dev/peps/pep-0020/

Reply:
Agreed, please change these occurrences to the more explicit implementation in a new PR.